app/bot.py

```diff
@@ -26,4 +27,93 @@ async def hello_function(ctx: SlashContext) -> None:
     await ctx.send("Hello world")
 
 
+@slash_command(
+    name="calculate",
+    description="Calculate a mathematical expression",
+    options=[
+        SlashCommandOption(
+            name="expression",
+            description="The mathematical expression to calculate",
+            type=OptionType.STRING,
+            required=True,
+        ),
+    ],
+)
+async def calculate_function(ctx: SlashContext, expression: str) -> None:
+    """Calculate the given mathematical expression.
+
+    This function takes a string expression, evaluates it, and returns the result.
+    It handles potential calculation errors and unexpected exceptions.
+
+    Args:
+    ----
+        ctx (SlashContext): The context of the slash command.
+        expression (str): The mathematical expression to evaluate.
+
+    """
+    await ctx.defer()
+
+    expression_without_spaces = expression.replace(" ", "")
+
+    try:
+        result = evaluate_expression(expression_without_spaces)
+        await ctx.send(f"The result of `{expression}` is: {result}", ephemeral=True)
+    except CalculationError as calc_error:
+        await ctx.send(f"Calculation error: {calc_error}", ephemeral=True)
+    except Exception as unexpected_error:  # noqa: BLE001
+        await ctx.send(f"An unexpected error occurred: {unexpected_error}", ephemeral=True)
+
+
+@slash_command(
+    name="calculate_info",
+    description="Get the list of allowed functions, constants, and operators",
+)
+async def info_function(ctx: SlashContext) -> None:
+    """Provide information about allowed functions, constants, and operators."""
+    functions_info = [
+        "`root(x, n)` - nth root of x",
+        "`log(x, base)` - logarithm of x with given base (default is e)",
+        "`ln(x)` - natural logarithm of x",
+        "`exp(x)` or `e^x` - exponential of x",
+        "`sin(x)`, `cos(x)`, `tan(x)` - trigonometric functions",
+        "`asin(x)`, `acos(x)`, `atan(x)` - inverse trigonometric functions",
+        "`sinh(x)`, `cosh(x)`, `tanh(x)` - hyperbolic functions",
+        "`asinh(x)`, `acosh(x)`, `atanh(x)` - inverse hyperbolic functions",
+        "`abs(x)` - absolute value of x",
+        "`factorial(x)` or `fact(x)` or `x!` - factorial of x",
+        "`round(x)` - round x to the nearest integer",
+        "`ceil(x)` - ceiling of x",
+        "`floor(x)` - floor of x",
+        "`sec(x)` - secant of x",
+        "`csc(x)` - cosecant of x",
+        "`cot(x)` - cotangent of x",
+        "`sqrt(x)` - square root of x",
+    ]
```

Comment:
All the operations could be implemented as a sub-command. That would make the execution easier, because the function names would be suggested by discord and brackets don't have to be typed.